src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,173 @@
-                && !SYSTEM_ERR_ALT.equals(logLocation)
-                && !ConsoleAppender.Target.SYSTEM_OUT.toString().equals(logLocation)
-                && !SYSTEM_OUT_ALT.equals(logLocation)) {
-
-            target = "FileLogger";
-
-            RollingFileAppender fa = RollingFileAppender.newBuilder()
-                .setConfiguration(config)
-                .withName(target)
-                .withLayout(layout)
-                .withFileName(logLocation)
-                .withFilePattern(logLocation + ".%d")
-                .withPolicy(SizeBasedTriggeringPolicy.createPolicy("5MB"))
-                .withStrategy(DefaultRolloverStrategy.newBuilder().withMax("1").build())
-                .build();
-
-            fa.start();
-            config.addAppender(fa);
-            ctx.getRootLogger().addAppender(config.getAppender(fa.getName()));
-
-            log.info("File Handler set");
-        } else {
-
-            if (logLocation != null
-                    && (ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
-                        || SYSTEM_ERR_ALT.equals(logLocation))) {
-
-                ConsoleAppender console = (ConsoleAppender)config.getAppender("CONSOLE");
-                console.stop();
-                config.getRootLogger().removeAppender("CONSOLE");
-                ctx.updateLoggers();
-
-                ConsoleAppender ca = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(logLocation)
-                    .setTarget(ConsoleAppender.Target.SYSTEM_ERR)
-                    .withLayout(layout)
-                    .build();
-
-                ca.start();
-                config.addAppender(ca);
-                ctx.getRootLogger().addAppender(config.getAppender(ca.getName()));
+            @Override
+            public synchronized String format(LogRecord lr) {
+                return String.format(format,
+                    dateFormatter.format(new Date()).toString(),
+                    LogLevel.fromJulLevel(lr.getLevel()).toString(),
+                    simpleClassName(lr.getSourceClassName()),
+                    lr.getMessage()
+                );
+            }
+        };
+
+        // prepare the different handlers
+        // --
+
+        ConsoleHandler stdoutHandler = null;
+        ConsoleHandler stderrHandler = null;
+        FileHandler fileHandler = null;
+
+        // the logLocation isn't always containing a file, it is sometimes
+        // referring to a standard output. We want to create a FileHandler only
+        // if the logLocation is a file on the FS.
+        if (logLocation != null && logLocation.length() > 0) {
+            if (!isStdOut(logLocation) && !isStdErr(logLocation)) {
+                // file logging
+                try {
+                    // maximum one 5MB file
+                    fileHandler = new FileHandler(logLocation, MAX_FILE_SIZE, FILE_COUNT);
```

Comment:
If I'm reading the javadocs right, I think this does a different thing than we did before: now, with `FILE_COUNT` set to 1, we only write to one file and don't really rotate it (see https://docs.oracle.com/javase/7/docs/api/java/util/logging/FileHandler.html#FileHandler(java.lang.String,%20int,%20int), I guess the logger just writes to the same file again once it's reached the max size (not sure).
Also the `logLocation` that we pass is the absolute path to the log file, but instead this expects a pattern, see the top of the docs page: https://docs.oracle.com/javase/7/docs/api/java/util/logging/FileHandler.html

Reply:
I think from reading this, it's a bit ambiguous in the docs about what happens here but the behavior might match:

From the intro to the class
```
For a rotating set of files, as each file reaches a given size limit, it is closed, rotated out, and a new file opened. Successively older files are named by adding "0", "1", "2", etc. into the base filename.
```
This makes me think that the code should maintain a single rotating log file like it did previously but I could be wrong (I'm going to test this locally to see what happens)

Reply:
Fixed now that you've changed `FILE_COUNT` to `2`